src/Model/Model.ts

```diff
@@ -221,43 +226,70 @@ export class Model {
     }
   }
 
-  public get id() {
-    return this.$currentData._id;
-  }
-
-  public get isDirty(): boolean {
-    return Object.keys(this.$dirty()).length > 0;
+  protected async $ensureCollection() {
+    if (!Model.$database) {
+      throw new Error('Model should only be accessed from IoC container');
+    }
+    if (this.$collection !== null) return;
+    const collectionName = computeCollectionName(this.constructor.name);
```

Comment:
The collection name can be customized in the constructor (`static collectionName = 'something'`). This doesn't take it into account.
We should add a test with a `class Something extends Model { static collectionName = 'somethingElse' }` that ends up calling this method to make sure it works.